tensorflow/core/common_runtime/rendezvous_mgr.cc

```diff
@@ -65,10 +65,10 @@ void IntraProcessRendezvous::SameWorkerRecvDone(
     StatusCallback done) {
   // Do a quick copy (sharing the underlying buffer) if both tensors
   // are on host memory.
-  const bool src_host =
-      (send_args.alloc_attrs.on_host() || parsed.src.type == "CPU");
-  const bool dst_host =
-      (recv_args.alloc_attrs.on_host() || parsed.dst.type == "CPU");
+  const bool src_host = (send_args.alloc_attrs.on_host() ||
+                         parsed.src.type == "CPU" || parsed.src.type == "SYCL");
```

Comment:
Yeah, you probably need to have a device-specific allocator.

The change here is basically saying that from the perspective of the framework, whenever you try to copy to a SYCL device, the destination should be treated as "host memory" (e.g., on the CPU).  This probably isn't what you want: you likely want the data to be copied into "device memory" which will have been allocated using a custom device allocator for SYCL.

In the CUDA case, when you have a send/recv pair between a CPU and GPU subgraph, you would want to transfer from CPU memory into GPU memory allocated via cuMalloc.  Our GPU allocator implementation for CUDA does this allocation, so you can see that our existing GPUDevice uses a GPUBFCAllocator implementation to allocate device memory via cuMalloc.

There are some cases where a kernel is written such that some of its inputs are not really needed in "device memory" -- they are more like 'control' bits (e.g., the index arguments to reshape).  Those are marked as "HostMemory" in their kernel registrations.  So this piece of code checks whether the source and destination are in 'host/cpu' memory.  If so, no copy needs to be made because the memory address spaces are the same.  Does that make sense?